mwdb/model/karton.py

```diff
@@ -85,6 +86,27 @@ def processing_in(self):
     def get(cls, analysis_id):
         return db.session.query(KartonAnalysis).filter(KartonAnalysis.id == analysis_id)
 
+    @classmethod
+    def get_or_create(cls, analysis_id, initial_object):
+        analysis = cls.get(analysis_id).first()
+
+        if analysis is not None:
+            return analysis, False
+
+        db.session.begin_nested()
+        analysis = KartonAnalysis(id=analysis_id, objects=[initial_object])
+        is_new = False
+        try:
+            db.session.add(analysis)
+            db.session.commit()
+            is_new = True
+        except IntegrityError:
+            db.session.rollback()
+            analysis = cls.get(analysis_id).first()
```

Comment:
Is this in case we have a race condition? If yes, maybe add a short comment like
 ```suggestion
             analysis = cls.get(analysis_id).first()  # Avoid TOCTOU exceptions
```